src/java.base/share/classes/jdk/internal/classfile/impl/BufWriterImpl.java

```diff
@@ -78,6 +79,14 @@ public void setThisClass(ClassEntry thisClass) {
         this.thisClass = thisClass;
     }
 
+    public int getMajorVersion() {
+        return majorVersion;
+    }
+
+    public void setMajorVersion(int majorVersion) {
```

Comment:
We should ensure the version is not changed once writing has already happened, and the constant pool builder should have access to the major version as well to prevent writing of invalid entries (like condy before java 11)

Reply:
the version field is now final